FWCore/Services/src/SiteLocalConfigService.cc

```diff
@@ -192,19 +191,29 @@ namespace edm {
       if (aDataCatalog.site == aDataCatalog.storageSite) {
         //it is a site (no defined subSite), use local path given in SITECONFIG_PATH
         if (aDataCatalog.subSite.empty())
-          filename_storage = siteconfig_path + "/storage.json";
+          filename_storage = siteconfig_path;
         //it is a subsite, move one level up
         else
-          filename_storage = siteconfig_path + "/../storage.json";
+          filename_storage = siteconfig_path + "/..";
       } else {  //cross site
         //it is a site (no defined subSite), move one level up
         if (aDataCatalog.subSite.empty())
-          filename_storage = siteconfig_path + "/../" + aDataCatalog.storageSite + "/storage.json";
+          filename_storage = siteconfig_path + "/../" + aDataCatalog.storageSite;
         //it is a subsite, move two levels up
         else
-          filename_storage = siteconfig_path + "/../../" + aDataCatalog.storageSite + "/storage.json";
+          filename_storage = siteconfig_path + "/../../" + aDataCatalog.storageSite;
+      }
+      filename_storage = filename_storage.string() + "/storage.json";
+      try {
+        filename_storage = std::filesystem::canonical(filename_storage);
+      } catch (std::exception &e) {
+        cms::Exception ex("SiteLocalConfigService");
+        ex << "Fail to convert path to the storage description, " << filename_storage.string()
+           << ", to the canonical absolute path"
+           << ". Path exists?";
```

Comment:
Would it be useful to include `e.what()` in the exception message?

Reply:
I see ```e.what()``` just says "std::exception" when path does not exist, which ```try``` catches. I see no benefit of include ```e.what()```

Reply:
Ok.